WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadStarter.kt

```diff
@@ -61,6 +63,13 @@ open class UploadStarter @Inject constructor(
     private val job = Job()
     private val MAXIMUM_AUTO_INITIATED_UPLOAD_RETRIES = 10
 
+    /**
+     * When the app comes to foreground both `queueUploadFromAllSites` and `queueUploadFromSite` are invoked.
+     * The problem is that they can run in parallel and `uploadServiceFacade.isPostUploadingOrQueued(it)` might return
+     * out-of-date result and a same post is added twice. This mutex serilizes
+     */
+    private val mutex = Mutex()
```

Comment:
Not important: Could we perhaps use [`newSingleThreadContext`](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/new-single-thread-context.html)?

Reply:
I wanted to use it as well, but then I found out that it's marked as experimental so I decided against it. Wdyt?

Reply:
For reference, we discussed this in a call with @shiki  and @malinajirka - we agreed that the real problem resides in the management of the upload queue and using a mutex here is a temporary fix for that problem. We'll eventually fix the upload queue make it concurrent-proof - it's out of scope for this PR and project.

Reply:
TIL `Executors.newSingleThreadExecutor().asCoroutineDispatcher()` thanks! ;) 

As you mentioned I'd take care of this issue in another PR so things don't get out of hands.